controllers/clustercache/cluster_accessor.go

```diff
@@ -298,7 +301,9 @@ func (ca *clusterAccessor) Connect(ctx context.Context) (retErr error) {
 // Disconnect disconnects a connection to the workload cluster.
 func (ca *clusterAccessor) Disconnect(ctx context.Context) {
 	log := ctrl.LoggerFrom(ctx)
-
+	defer func() {
```

Comment:
Maybe let's move this directly above l.315?

Basically in l.307-l.310 nothing changes so we also don't have to change the metric

Not super important, but also to keep this similar to how we handle the metric in the Connect method